tests/run/nonecheck.pyx

```diff
@@ -190,3 +190,16 @@ def test_memslice_slice_assign2(double[:] buf):
     TypeError: Cannot slice None memoryview slice
     """
     buf[:] = buf[::-1]
+
+cpdef non_none_arg_hint(a: MyClass):
+    return a
+    
+def test_call_non_none_arg_hint_with_none():
+    """
+    >>> test_call_non_none_arg_hint_with_none()
```

Comment:
The test should also include a call to `non_none_arg_hint` from Python (i.e. in the doctest) just to show that they're behaving the same (since I think the bug report is that they're not behaving the same)